connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/SynchronizationTest.java

```diff
@@ -461,9 +461,7 @@ private static ThreadFactory threadFactoryWithNamedThreads(String threadPrefix)
         return r -> {
             // This is essentially Executors.defaultThreadFactory except with
             // custom thread names so in order to filter by thread names when debugging
-            SecurityManager s = System.getSecurityManager();
-            Thread t = new Thread((s != null) ? s.getThreadGroup() :
-                Thread.currentThread().getThreadGroup(), r,
+            Thread t = new Thread(Thread.currentThread().getThreadGroup(), r,
```

Comment:
Java 17 is deprecated the SecurityManager, see [doc](https://openjdk.org/jeps/411)